doc/rofilib/configuration.rst

```diff
@@ -1,11 +1,52 @@
 Configuration
 =============
 
-Here you are supposed to say the general idea behind the configuration
-representation and also list its capabilities. (e.g., you create and instance,
-put modules, then you have to compute the matrix, etc.)
+Configuration is a way of representing any bot in an universal manner. It
+contains all building-blocks of the RoFI platform such as various modules
+and joints. Using these blocks, you can specify every rofibot and work
+with it. Configuration computes positions of every component within the bot
+and provides useful functions for manipulation with such rofibot, including
+some validity checks (e.g., collision checks).
 
-Feel free to add code examples.
+Using the configuration is pretty straightforward. The main class is
+:cpp:class:`Rofibot <rofi::configuration::Rofibot>` for which you create an
+instance and add all the modules you have within one bot. You have to connect
+them appropriately and if you want to use absolute positions, you have to
+fix one of the bot's components in space (otherwise, the configuration cannot
+figure out its coordinates because everything is kept relative).
+
+For example, let us create a small bot with two universal modules.
+
+.. code-block:: cpp
+
+    #include <configuration/rofibot.hpp>
```

Comment:
I would rather put it as a comment since includes and code can't be on the same level

Reply:
I would prefer this way (from `Result`)

```cpp
    #include <atoms/util.hpp> // for `overloaded`

    // ...

    auto correctAnswer = atoms::Result< int >::value( 10 );
```

so I would just add a comment in between.